src/components/audio-params.ts

```diff
@@ -28,16 +28,16 @@ export enum PanningModelType {
 }
 
 export interface AudioSettings {
-  audioType: AudioType;
-  distanceModel: DistanceModelType;
-  panningModel: PanningModelType;
-  rolloffFactor: number;
-  refDistance: number;
-  maxDistance: number;
-  coneInnerAngle: number;
-  coneOuterAngle: number;
-  coneOuterGain: number;
-  gain: number;
+  audioType?: AudioType;
+  distanceModel?: DistanceModelType;
+  panningModel?: PanningModelType;
+  rolloffFactor?: number;
+  refDistance?: number;
+  maxDistance?: number;
+  coneInnerAngle?: number;
+  coneOuterAngle?: number;
+  coneOuterGain?: number;
+  gain?: number;
```

Comment:
I don't think these properties should be optional. 

If you made these optional so that each property of the `params` sent to `inflateAudioParams` can be optional, then those `params` should be a different type. 

Dom mentioned some typescript utilities that will let us handle this without duplicating a bunch of lines (and risking mistakes): https://github.com/mozilla/hubs/pull/5933#discussion_r1105232292

Reply:
I understand how `Partial` works but I'm not sure why here is the correct option in this case. What's the difference in this case of making all properties optional or using `Partial<AudioSettings>`? Those settings can be optional everywhere so this just adds the need of making it `Partial` everywhere where we use `AudioSettings`.

Reply:
Defaults should not be `Partial`, and the reducer in `update-audio-settings` should end up with something that is also not a `Partial`